src/components/questionary/questionaryComponents/SampleDeclaration/QuestionSampleDeclarationForm.tsx

```diff
@@ -110,66 +113,59 @@ export const QuestionSampleDeclarationForm: FC<QuestionFormProps> = (props) => {
             </FormControl>
 
             {features.get(FeatureId.RISK_ASSESSMENT)?.isEnabled && (
-              <FormControl fullWidth>
-                <InputLabel htmlFor="config.esiTemplateId">
-                  ESI template name
-                </InputLabel>
-                <Field
-                  name="config.esiTemplateId"
-                  id="config.esiTemplateId"
-                  type="text"
-                  component={Select}
-                  data-cy="esi-template-id"
-                >
-                  {esiTemplates.length ? (
-                    esiTemplates.map((template) => {
-                      return (
-                        <MenuItem
-                          value={template.templateId}
-                          key={template.templateId}
-                        >
-                          {template.name}
-                        </MenuItem>
-                      );
-                    })
-                  ) : (
-                    <MenuItem value="noTemplates" key="noTemplates" disabled>
-                      No active templates
-                    </MenuItem>
-                  )}
-                </Field>
-              </FormControl>
+              <FormikUIAutocomplete
+                name="config.esiTemplateId"
+                label="ESI template name"
+                noOptionsText="No active templates"
+                items={ESITemplateOptions}
+                InputProps={{ 'data-cy': 'esi-template-id' }}
+                TextFieldProps={{ margin: 'none' }}
+              />
             )}
+          </TitledContainer>
 
-            <Field
-              name="config.addEntryButtonLabel"
-              id="Add-button-label-Input"
-              label="Add button label"
-              placeholder='(e.g. "add new")'
-              type="text"
-              component={TextField}
-              fullWidth
-              data-cy="addEntryButtonLabel"
-            />
+          <TitledContainer label="Constraints">
             <Field
               name="config.minEntries"
               id="Min-Input"
               label="Min entries"
               placeholder="(e.g. 1, leave blank for unlimited)"
-              type="text"
+              type="number"
+              inputProps={{ min: 0 }}
               component={TextField}
               fullWidth
               data-cy="min-entries"
+              // NOTE: This is needed to prevent sending empty string when there is no value
+              onChange={({
```

Comment:
Should we move the event handler out?

Reply:
This is a bit more complex to do because of setFieldValue not available in outside form. I should send it as an argument of the handler or extract from useForm hook. But for now I think this is a bit better as we decided not to go with some universal handler cause this is cleaner.